src/main/java/com/commercetools/sync/categories/helpers/CategoryReferenceResolver.java

```diff
@@ -111,45 +110,32 @@ public CategoryReferenceResolver(@Nonnull final CategorySyncOptions options,
     @Nonnull
     CompletionStage<CategoryDraftBuilder> resolveParentReference(@Nonnull final CategoryDraftBuilder draftBuilder) {
         try {
-            return getParentCategoryKey(draftBuilder, options.shouldAllowUuidKeys())
+            return getParentCategoryKey(draftBuilder)
                 .map(parentCategoryKey -> fetchAndResolveParentReference(draftBuilder, parentCategoryKey))
-                .orElseGet(() -> CompletableFuture.completedFuture(draftBuilder));
+                .orElseGet(() -> completedFuture(draftBuilder));
         } catch (ReferenceResolutionException referenceResolutionException) {
-            return CompletableFutureUtils
-                .exceptionallyCompletedFuture(referenceResolutionException);
+            return exceptionallyCompletedFuture(referenceResolutionException);
         }
     }
 
-    /**
-     * Given a {@link CategoryDraftBuilder} and a {@code parentCategoryKey} this method fetches the actual id of the
-     * category corresponding to this key, ideally from a cache. Then it sets this id on the parent reference
-     * id. If the id is not found in cache nor the CTP project, the resultant draft builder
-     * would remain exactly the same as the passed category draft (without parent reference resolution).
-     *
-     * @param draftBuilder the category draft builder to accept resolved references values.
-     * @param parentCategoryKey the parent category key of to resolve it's actual id on the draft.
-     * @return a {@link CompletionStage} that contains as a result the same {@code draftBuilder} category draft builder
-     *         instance with resolved parent category references or an exception.
-     */
     @Nonnull
-    private CompletionStage<CategoryDraftBuilder> fetchAndResolveParentReference(
-            @Nonnull final CategoryDraftBuilder draftBuilder,
-            @Nonnull final String parentCategoryKey) {
-        return categoryService.fetchCachedCategoryId(parentCategoryKey)
-            .thenApply(resolvedParentIdOptional -> resolvedParentIdOptional
-                .map(resolvedParentId ->
-                    draftBuilder.parent(Category.referenceOfId(resolvedParentId).toResourceIdentifier()))
-                .orElse(draftBuilder));
+    public static Optional<String> getParentCategoryKey(@Nonnull final CategoryDraftBuilder draftBuilder)
+        throws ReferenceResolutionException {
+        return getParentCategoryKey(draftBuilder.getParent(), draftBuilder.getKey());
+    }
+
+    @Nonnull
+    public static Optional<String> getParentCategoryKey(@Nonnull final CategoryDraft draft)
+        throws ReferenceResolutionException {
+        return getParentCategoryKey(draft.getParent(), draft.getKey());
     }
 
     /**
      * Given a category parent resource identifier, if it is not null the method validates the id field value. If it is
      * not valid, a {@link ReferenceResolutionException} will be thrown. The validity checks are:
```

Comment:
parent resource key ...

Reply:
not sure what you mean here? It is a resource Identifier that's supplied, not just key.

Reply:
ok